folioreader/src/main/java/com/folioreader/ui/folio/adapter/FolioPageFragmentAdapter.java

```diff
@@ -38,6 +48,51 @@ public FolioPageFragmentAdapter(FragmentManager fragmentManager, List<Link> spin
         this.mBookId = bookId;
         this.publication = publication;
         fragments = new ArrayList<>(Arrays.asList(new Fragment[mSpineReferences.size()]));
+        loadAllFragments(mSpineReferences);
+    }
+
+    private void loadAllFragments(List<Link> references) {
```

Comment:
is this going to impact performance instantiating all of these fragments right away?

Reply:
I didn't notice any performance drawbacks